napari/layers/image/experimental/_octree_loader.py

```diff
@@ -233,62 +292,58 @@ def _get_coverage(
         if ideal_chunk.in_memory and ideal_chunk in drawn_set:
             return [ideal_chunk]
 
-        # Get alternates for this chunk, from other levels.
-        family = self._get_family(ideal_chunk)
+        # If not, get alternates for this chunk, from other levels.
 
-        ideal_level_index = ideal_chunk.location.level_index
+        # If the ideal chunk is in memory then we'll want to draw that one
+        # too though
+        if ideal_chunk.in_memory:
+            best_in_memory_chunk = [ideal_chunk]
+        else:
+            best_in_memory_chunk = []
 
-        # For levels below the ideal level, we only keep an alternate if
-        # it's already being drawn. This is usually when zooming out. The
-        # alternates are "too small" but still look fine on screen.
-        #
-        # For levels above the ideal level, we will load and draw them. We
-        # even sort so they get loaded and drawn *before* the ideal chunk.
-        #
-        # We do this because they provide coverage very quickly, and the
-        # best user experience is to see imagery quickly even if not at the
-        # ideal level.
-        def keep_chunk(chunk) -> bool:
-            lower_level = chunk.location.level_index < ideal_level_index
-
-            if lower_level:
-                return chunk in drawn_set
-
-            return True  # Keep all higher level chunks.
-
-        keep = [chunk for chunk in family if keep_chunk(chunk)]
-
-        return keep
-
-    def _get_family(self, ideal_chunk: OctreeChunk) -> List[OctreeChunk]:
-        """Return chunks below and above this ideal chunk.
-
-        Parameters
-        ----------
-        ideal_chunk : OctreeChunk
-            Get children and parents of this chunk.
-
-        Returns
-        -------
-        List[OctreeNode]
-            Parents and children we should load and/or draw.
-        """
-        # Get any direct children which are in memory. Do not create
+        # First get any direct children which are in memory. Do not create
         # OctreeChunks or use children that are not already in memory
         # because it's better to create and load higher levels.
         children = self._octree.get_children(
             ideal_chunk, create=False, in_memory=True
         )
 
-        # Get the parent and maybe more distant ancestors. Even if we have
-        # all four children, we still consider loading and drawing these
-        # because they will provide more coverage. They will cover the
-        # ideal chunk plus more.
+        # Only keep the children which are already drawn, as drawing is
+        # expensive don't want to draw them unnecessarily.
+        children = [chunk for chunk in children if chunk in drawn_set]
+
+        # If all four children are in memory and are already drawn just return them
+        # as they will cover the whole chunk.
+        if len(children) == 4:
```

Comment:
Do we have a concept of dimensionality here? if so, even though there's lots more work to do, how about changing this to `if len(children) == 2**ndim:` for future improvements?

Reply:
(If not, then ignore and resolve this thread ;)

Reply:
We're very much only 2D right now, lot's of "row/ column" references, but I made this `2**ndim` and added a comment for future reference. TBH though, I'm not sure the same "coverage" logic would apply in 3D, but i don't see harm

Reply:
I like it now, a little more clear why it is four